feat: implement Vue.js template

Bina-man opened the pull request:
- Add Vue router setup with sample pages
- Configure Prettier for code formatting
- Set up project structure
- Implement dashboard and sample pages
- Add API client structure
- Add navigation component
- Add composables and layouts

Bina-man committed:
feat: implement Vue.js template

- Add Vue router setup with sample pages
- Configure Prettier for code formatting
- Set up project structure
- Implement dashboard and sample pages
- Add API client structure
- Add navigation component
- Add composables and layouts

Bina-man committed:
Update .gitignore and add encrypted env file

Bina-man commented:
![Uploading Screenshot 2025-02-07 at 10.32.12 in the morning.png…]()

Bina-man commented:
<img width="1712" alt="Screenshot 2025-02-07 at 10 32 18 in the morning" src="https://github.com/user-attachments/assets/b98094e6-2cd4-42e9-ae7e-eb5806dbfd0f" />
![Uploading Screenshot 2025-02-07 at 10.32.24 in the morning.png…]()

hillaryke commented on src/api/client.ts:
```diff
@@ -0,0 +1,40 @@
+import axios from 'axios'
+import type { User, Stats, Activity } from './types'
+
+const apiClient = axios.create({
```
Since most of the projects created have been using Nuxt, which has built in fetch utilities, we can consider using that over axios. We can refer to platform-web and chatbot-web-general for Nuxt best practices and setup

hillaryke commented on src/main.ts:
```diff
@@ -0,0 +1,8 @@
+import './style.css'
+
+const app = createApp(App)
+app.use(router)
```
Nuxt has page routes funcationality both for ui pages and server api routes. We can consider implementing that

Bina-man replied:
Thanks for the review! I've implemented the suggested changes to use Nuxt's built-in fetch utilities instead of axios. The API client has been updated to use `useFetch`, maintaining the same functionality but aligning with Nuxt best practices. 

Let me know if you'd like me to make any additional adjustments!

hillaryke commented:
I'm including the message here for reference that I had sent on ford channel on Discord concerning the consideration of using Nuxt 3 over Vue. I have made minor modifications to include only relevant details for this project

> 
> I noticed a few areas where we could enhance the setup to match our best practices and be up to date with regard to Nuxt recommendations:
> 
> Technical Suggestions:
> 1. Since we're starting fresh, we could upgrade to Nuxt 3 which would give us:
>    - Better TypeScript support out of the box (meaning we'd use `nuxt.config.ts` over `nuxt.config.js` and `vite.config.js`)
>    - Built-in fetch utilities i.e. ($fetch or useFetch) (we can remove axios)
>    - Improved performance and smaller bundle size
> 
> 2. For the project structure, we might want to:
>    - Sdd some key Nuxt directories (layouts, middleware, plugins) that will be needed as features grow
>    - Follow our design system component structure (/components/d/s) from platform-web
> 
> 3. For development and performance:
>    - Configure auto-imports and lazy loading (this would eliminate importing files manually) (refer to mcp-chatbot's `nuxt.config.ts`)
>    - Setup Vitest for testing (this is recommended for Nuxt 3)

Bina-man committed:
refactor: update API client to use Nuxt's useFetch

- Replace axios with Nuxt's built-in useFetch utility
- Maintain existing API structure and typings
- Improve type safety with proper generics

Bina-man committed:
chore: add Nuxt and Vitest configuration files

Bina-man committed:
feat: add base app layout with navigation and theme

Bina-man committed:
feat: add sample page with UI components

Bina-man committed:
feat: add dashboard with stats and activity table

Bina-man committed:
refactor: remove Vue CLI specific files for Nuxt migration

Bina-man committed:
feat: add Nuxt directory structure and base pages

Bina-man committed:
chore: update gitignore and package dependencies for Nuxt
